src/numbers.rs

```diff
@@ -0,0 +1,1423 @@
+        if self.0 < 0 {
+            Integer(-self.0)
+        } else {
+            *self
+        }
+    }
+
+    pub const fn wrapping_abs(&self) -> Self {
+        if self.0 < 0 {
+            Integer(-self.0)
+        } else {
+            *self
+        }
+    }
+
+    pub const fn wrapping_add(&self, other: &Self) -> Self {
+        let result = self.0.wrapping_add(other.0);
+        Integer(result)
+    }
+
+    pub const fn wrapping_add_unsigned(&self, rhs: u128) -> Self {
+        let result = self.0.wrapping_add(rhs as i128);
+        Integer(result)
+    }
+
+    pub const fn wrapping_div(&self, other: &Self) -> Self {
+        if other.0 == 0 {
+            return Integer(i128::MAX);
+        }
+        Integer(self.0.wrapping_div(other.0))
+    }
+
+    pub const fn wrapping_div_euclid(&self, other: &Self) -> Self {
+        if other.0 == 0 {
+            return Integer(i128::MAX);
+        }
+        Integer(self.0.wrapping_div_euclid(other.0))
+    }
+
+    pub const fn wrapping_mul(&self, other: &Self) -> Self {
+        let result = self.0.wrapping_mul(other.0);
+        Integer(result)
+    }
+
+    pub const fn wrapping_neg(&self) -> Self {
+        let result = self.0.wrapping_neg();
+        Integer(result)
+    }
+
+    pub const fn wrapping_pow(&self, exp: u32) -> Self {
+        let result = self.0.wrapping_pow(exp);
+        Integer(result)
+    }
+
+    pub const fn wrapping_rem(&self, other: &Self) -> Self {
+        if other.0 == 0 {
+            return Integer(i128::MAX);
+        }
+        Integer(self.0.wrapping_rem(other.0))
+    }
+
+    pub const fn wrapping_rem_euclid(&self, other: &Self) -> Self {
+        if other.0 == 0 {
+            return Integer(i128::MAX);
+        }
+        Integer(self.0.wrapping_rem_euclid(other.0))
+    }
+
+    pub const fn wrapping_shl(&self, rhs: u32) -> Self {
+        let result = self.0.wrapping_shl(rhs);
+        Integer(result)
+    }
+
+    pub const fn wrapping_shr(&self, rhs: u32) -> Self {
+        let result = self.0.wrapping_shr(rhs);
+        Integer(result)
+    }
+
+    pub const fn wrapping_sub(&self, other: &Self) -> Self {
+        let result = self.0.wrapping_sub(other.0);
+        Integer(result)
+    }
+
+    pub const fn wrapping_sub_unsigned(&self, rhs: u128) -> Self {
+        let result = self.0.wrapping_sub(rhs as i128);
+        Integer(result)
+    }
+
+    /// Returns the cbor info byte for the integer.
+    pub fn cbor_info(&self) -> Result<u8, crate::error::NumbersError> {
+        let length = self.serialized_len();
+        match length {
+            1 => Ok((self.0 & 0x1F) as u8),
+            2 => Ok(0x18),
+            3 => Ok(0x19),
+            5 => Ok(0x1A),
+            9 => Ok(0x1B),
+            _ => Err(crate::error::NumbersError::InvalidCBORIntLength { length }),
+        }
+    }
+
+    // /// Returns whether the inner value of integer can fit into the target type.
+    // pub fn fits_into<T: IntegerType>(&self) -> bool {
```

Comment:
Should this be uncommented or removed?

Reply:
Woops! Yep. definitely need to finish implementing that! Should be on the next commit push.